pkg/build/strategies/sti/sti.go

```diff
@@ -467,22 +469,26 @@ func (b *STI) Execute(command string, user string, config *api.Config) error {
 			return err
 		}
 		defer os.Remove(rmScript)
-		glog.V(5).Infof("Waiting for injected files to be copied into assemble container...")
 		opts.CommandOverrides = func(cmd string) string {
 			return fmt.Sprintf("while [ ! -f %q ]; do sleep 0.5; done; %s; result=$?; source %[1]s; exit $result",
 				"/tmp/rm-injections", cmd)
 		}
 		originalOnStart := opts.OnStart
 		opts.OnStart = func(containerID string) error {
+			defer close(injectionComplete)
 			if err != nil {
+				injectionError = err
 				return err
 			}
+			glog.V(2).Info("starting the injections uploading ...")
 			for _, s := range config.Injections {
 				if err := b.docker.UploadToContainer(s.SourcePath, s.DestinationDir, containerID); err != nil {
+					injectionError = util.HandleInjectionError(s, err)
 					return err
 				}
 			}
 			if err := b.docker.UploadToContainer(rmScript, "/tmp/rm-injections", containerID); err != nil {
+				injectionError = util.HandleInjectionError(api.InjectPath{SourcePath: rmScript, DestinationDir: "/tmp/rm-injections"}, err)
```

Comment:
I'm not sure but should we use `/tmp` for `DestinationDir` here? (As name implies?)